dask/dataframe/core.py

```diff
@@ -2152,6 +2181,99 @@ def quantile(df, q):
     return return_type(dsk, name3, df.name, new_divisions)
 
 
+def cov_corr(df, min_periods=None, corr=False, scalar=False):
+    """DataFrame covariance and pearson correlation.
+
+    Computes pairwise covariance or correlation of columns, excluding NA/null
+    values.
+
+    Parameters
+    ----------
+    df : DataFrame
+    min_periods : int, optional
+        Minimum number of observations required per pair of columns
+        to have a valid result.
+    corr : bool, optional
+        If True, compute the Pearson correlation. If False [default], compute
+        the covariance.
+    scalar : bool, optional
+        If True, compute covariance between two variables as a scalar. Only
+        valid if `df` has 2 columns.  If False [default], compute the entire
+        covariance/correlation matrix.
+    """
+    if min_periods is None:
+        min_periods = 2
+    elif min_periods < 2:
+        raise ValueError("min_periods must be >= 2")
+    prefix = 'corr' if corr else 'cov'
+    df = df._get_numeric_data()
+    name = '{0}-agg-{1}'.format(prefix, tokenize(df, min_periods, scalar))
+    if scalar and len(df.columns) != 2:
+        raise ValueError("scalar only valid for 2 column dataframe")
+    k = '{0}-chunk-{1}'.format(prefix, df._name)
+    dsk = dict(((k, i), (cov_corr_chunk, f, corr))
+               for (i, f) in enumerate(df._keys()))
+    dsk[(name, 0)] = (cov_corr_agg, list(dsk.keys()), df._pd, min_periods,
+                      corr, scalar)
+    dsk = merge(df.dask, dsk)
+    if scalar:
+        return Scalar(dsk, name)
+    return DataFrame(dsk, name, df._pd, (df.columns[0], df.columns[-1]))
+
+
+def cov_corr_chunk(df, corr=False):
+    """Chunk part of a covariance or correlation computation"""
+    mat = df.values
+    mask = np.isfinite(mat)
+    keep = np.bitwise_and(mask[:, None, :], mask[:, :, None])
+
+    x = np.where(keep, mat[:, None, :], np.nan)
+    sums = np.nansum(x, 0)
+    counts = keep.astype('int').sum(0)
+    cov = df.cov().values
+    dtype = [('sum', sums.dtype), ('count', counts.dtype), ('cov', cov.dtype)]
+    if corr:
+        m = np.nansum((x - sums/np.where(counts, counts, np.nan))**2, 0)
+        dtype.append(('m', m.dtype))
+
+    out = np.empty(counts.shape, dtype=dtype)
+    out['sum'] = sums
+    out['count'] = counts
+    out['cov'] = cov * (counts - 1)
+    if corr:
+        out['m'] = m
+    return out
```

Comment:
I thought about doing that, and could go either way. This made it a bit easier to work with both types of intermediates, and serializes slightly faster. The decision between the two was mostly arbitrary though.